UI/src/model/AssetsVerificationCollection.js

```diff
@@ -1,11 +1,31 @@
 (function(root) {
     root.AssetsVerificationCollection = function(backend) {
 
-        this.fetch = function(boundingBox, typeId) {
-            return backend.getVerificationInfo(boundingBox, typeId).then(function(result) {
-                var verified = result ? result.verified : false;
-                eventbus.trigger('verificationInfo:fetched', verified);
-            });
+        this.fetch = function (boundingBox, center, typeId, hasMunicipalityValidation) {
+            if(!hasMunicipalityValidation)
+                eventbus.trigger('verificationInfo:fetched', false);
+            else {
+                backend.getMunicipalityFromCoordinates(center[0], center[1], function (vkmResult) {
+                    if (vkmResult.kuntakoodi)
+                        backend.getVerificationInfo(vkmResult.kuntakoodi, typeId).then(function (result) {
+                            var verified = result ? result.verified : false;
+                            eventbus.trigger('verificationInfo:fetched', verified);
+                        });
+                    else {
+                        backend.getMunicipalityByBoundingBox(boundingBox).then(function (municipalityInfo) {
+                            if (municipalityInfo) {
+                                backend.getVerificationInfo(municipalityInfo, typeId).then(function (result) {
+                                    var verified = result ? result.verified : false;
+                                    eventbus.trigger('verificationInfo:fetched', verified);
+                                });
```

Comment:
instead of code duplicate, could be create this function

function getVerificationInfo(municipalityInfo, typeId) {
            backend.getVerificationInfo(municipalityInfo, typeId).then(function (result) {
                var verified = result ? result.verified : false;
                eventbus.trigger('verificationInfo:fetched', verified);
            });
        }